src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/PackageRepository.cs

```diff
@@ -57,7 +58,8 @@ public void OnPackageUninstalling(PackageCatalog p, PackageUninstallingEventArgs
             if (args.Progress == 0)
             {
                 //find apps associated with this package. 
-                var uwp = new UWP(args.Package);
+                var packageWrapper = PackageWrapper.GetWrapperFromPackage(args.Package);

```

Comment:
nit:  Just wanted to call out in most cases I would dependency inject some type of builder object here instead of calling a static method.  This allows tests to mock the builder to validate the buisness logic.  However in this case you can't construct, mock or shim the package* objects which makes unit testing these event handlers impossible in the first place :(

Reply:
That's true! It would have been a lot easier if implemented interfaces were not internal.